docs/LabVIEW_Debugging_Techniques.md

```diff
@@ -0,0 +1,95 @@
+### Debugging and Benchmarking
+
+Debugging and benchmarking is a normal process of code development. There are a variety of ways to debug and benchmark custom devices.
+
+#### LabVIEW Debugging Techniques
+
+Custom devices are written in LabVIEW code. Therefore it’s possible to develop, test and debug the code in LabVIEW, before running the niveristand-custom-device-wizard. In other words, you can use **[LabVIEW’s built-in Debugging Techniques](https://www.ni.com/en-us/support/documentation/supplemental/12/debugging-techniques-in-labview.html)** during development, and merge the LabVIEW code into the custom device framework after it matures.
+
+Since the custom device is one of many parts of the system definition, the behavior of the LabVIEW code within the custom device framework will likely differ from the stand-alone LabVIEW application, especially in regards to timing. As a result, you should benchmark the custom device inside of the VeriStand Engine.
+Once added to the system definition, custom devices have been fully integrated into VeriStand’s context. As a result, LabVIEW’s built-in Debugging Techniques are no longer available. Several techniques are available for debugging and benchmarking the custom device.
+ 
+#### Console Viewer
+
+A subcomponent of VeriStand RT Engine is the RT Console Viewer.
+You can install it to the execution host using Measurement and Automation Explorer. When installed, the component runs a small UDP daemon allowing the operator to view the console from a utility called the RT Console Viewer. You can access the RT Console viewer from the **VeriStand Editor » Tool Launcher » View Console**.
+
+![](images/ConsoleViewer1.jpg)
+
+**Figure: RT Console Viewer**
+
+The Console Viewer will show the system definition and the resulting CPU usage. The viewer is also useful for displaying debugging messages. The console viewer provides a periodic snapshot of utilization. CPU spikes and transients will probably be unobservable. If the system is very busy, it may not update the console viewer at all. You can use other debugging methods for a more accurate indication of resource utilization.
+
+As the name implies, the RT Console Viewer is only available on real-time targets. The RT Console Viewer is also available as a stand-alone add-on to LabVIEW Real-Time. See **[NI Developer Zone Tutorial: Remotely View Console Output of Real-Time Targets](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000x4TjCAI&l=en-US)** for more information.
+
+&nbsp;&nbsp;&nbsp; **Note:** You cannot use the **Console Viewer** on NI Linux Real-Time targets. Instead, connect your NI Linux Real-Time targets to a computer using a serial port to view the output.
```

Comment:
I would probably lead with this NOTE and make it more prominent since >= VS 2021 this is going to be the situation our customers will see a majority of the time (NI Linux RT targets)